examples/step-37/doc/results.dox

```diff
@@ -409,3 +411,229 @@ number of threads in the MPI_InitFinalize data structure in the main function,
 and set the MatrixFree::AdditionalData::tasks_parallel_scheme to
 partition_color to actually do the loop in parallel. This use case is
 discussed in step-48.
+
+<h4> Inhomogeneous Dirichlet boundary conditions </h4>
+
+The presented program assumes homogeneous Dirichlet boundary conditions. When
+going to non-homogeneous conditions, the situation is a bit more intricate. To
+understand how to implement such a setting, let us first recall how these
+arise in the mathematical formulation and how they are implemented in a
+matrix-based variant. In essence, an inhomogeneous Dirichlet condition sets
+some of the nodal values in the solution to given values rathern than
+determining them through the variational principles,
+@f{eqnarray*}
+u_h(\mathbf{x}) = \sum_{i=1}^{n_\mathrm{dofs}} \varphi_i(\mathbf{x}) u_i =
+\sum_{i=1}^{n_\mathrm{dofs}-n_\mathrm{dofs,D}} \varphi_i(\mathbf{x}) u_i +
+\sum_{i=n_\mathrm{dofs}-n_\mathrm{dofs,D}+1}^{n_\mathrm{dofs}} \varphi_i(\mathbf{x}) g_i,
+@f}
+where $u_i$ denotes the nodal values of the solution and $g_i =
+g(\mathbf{x}_i)$ denotes the interpolation of boundary values on the
+Dirichlet-constrained node points <i>i</i>. In this formula, where we have
+assumed that the degrees of freedom are numbered such that the ones subject to
+a Dirichlet condition are all appended to the end of the list of degrees of
+freedom. We then insert this solution representation into the weak form,
+e.g. the Laplacian shown above, and move the known quantities to the right
+hand side:
+@f{eqnarray*}
+(\varphi_i, u_h)_\Omega &=& (\varphi_i, f)_\Omega \Rightarrow \\
+\sum_{j=1}^{n_\mathrm{dofs}-n_\mathrm{dofs,D}}(\varphi_i,\varphi_j)_\Omega u_j =
+(\varphi_i, f)_\Omega
+-\sum_{j=n_\mathrm{dofs}-n_\mathrm{dofs,D}+1}^{n_\mathrm{dofs}} (\varphi_i,\varphi_j)_\Omega g_j.
```

Comment:
maybe add space after `_Omega` so that `g_j` is not that close and it's easier to read. sabe above for `u_j`.

Reply:
That's a good idea indeed, yes. I will adjust.